File: mo-common/mo-common-redis/src/main/java/ink/ckx/mo/common/redis/config/RedissonConfig.java
package ink.ckx.mo.common.redis.config;

import cn.hutool.core.text.CharSequenceUtil;
import lombok.Setter;
import org.redisson.Redisson;
import org.redisson.api.RedissonClient;
import org.redisson.config.Config;
import org.redisson.config.SingleServerConfig;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

/**
 * @author chenkaixin
 * @description
 * @since 2023/11/10
 */
@ConditionalOnProperty(prefix = "redisson", name = "address")
@ConfigurationProperties(prefix = "redisson")
@Configuration
@Setter
public class RedissonConfig {

    private String address;

    private String password;

    private Integer database;

    private Integer minIdle;

    @Bean
    public RedissonClient redissonClient() {
        Config config = new Config();
        SingleServerConfig singleServerConfig = config.useSingleServer();
        singleServerConfig.setAddress(address);
        singleServerConfig.setDatabase(database);
        singleServerConfig.setConnectionMinimumIdleSize(minIdle);
        if (CharSequenceUtil.isNotBlank(password)) {
            singleServerConfig.setPassword(password);
        }
        return Redisson.create(config);
    }
}

File: mo-common/mo-common-redis/src/main/kotlin/ink/ckx/mo/common/redis/config/RedissonConfig.kt
package ink.ckx.mo.common.redis.config

import cn.hutool.core.util.StrUtil
import org.redisson.Redisson
import org.redisson.api.RedissonClient
import org.redisson.config.Config
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

/**
 * @author chenkaixin
 * @description
 * @since 2023/11/10
 */
@ConditionalOnProperty(prefix = "redisson", name = ["address"])
@Configuration
class RedissonConfig {

    @Bean
    @ConfigurationProperties(prefix = "redisson")
    fun redissonConfigProperties(): RedissonConfigProperties {
        return RedissonConfigProperties()
    }

    @Bean
    fun redissonClient(properties: RedissonConfigProperties): RedissonClient {
        val config = Config()
        val singleServerConfig = config.useSingleServer()
        singleServerConfig.address = properties.address
        singleServerConfig.database = properties.database
        singleServerConfig.connectionMinimumIdleSize = properties.minIdle
        if (StrUtil.isNotBlank(properties.password)) {
            singleServerConfig.password = properties.password
        }
        return Redisson.create(config)
    }
}

data class RedissonConfigProperties(
    var address: String = "redis://127.0.0.1:6379",
    var password: String = "123456",
    var database: Int = 0,
    var minIdle: Int = 1,
)